watchtower/wtclient/client.go

```diff
@@ -760,8 +760,52 @@ func (c *TowerClient) backupDispatcher() {
 	for {
 		switch {
 
-		// No active session queue and no additional sessions.
+		// No active session queue and no additional sessions,
+		// so we'll need to request a new one.
 		case c.sessionQueue == nil && len(c.candidateSessions) == 0:
+
+			// Wait until we have at least one candidate tower.
+			// with which to request a session.
+			if c.candidateTowers.IsEmpty() {
```

Comment:
hmmm so I think instead of adding another layer of complexity on this level, we should move this down to the session negotiator. The session negotiator has access to the tower iterator already - so perhaps it can check if it is empty and then if it is, it can wait for the tower iterator to signal (via a new method on the tower iterator itself) to it when it has received a new tower candidate

Reply:
You've been putting in work on the tower lately, so your guess is certainly better than mine. I gave this a go, but in so doing, I did notice one aspect that is hard to recreate if pushed one layer deeper into the session negotiator. Namely when we reconnect to a tower seen previously, we will negotiate new sessions with said tower ahead of reusing any existing sessions. 

This is a consequence of blocking on empty candidate list "during" session negotiation vs. the PR's current approach of the client blocking ***prior to*** requesting a new session be negotiated. The latter enables the client to check for and re-use any existing non-exhausted sessions previously opened with the tower.

The existing but non-exhausted sessions will STILL be used eventually as they are pulled out of DB and added to our list of candidate sessions. As a result, I don't think passing on this aspect would negatively impact disk usage with poorly utilized sessions. Plus, I'm pretty sure this is actually how the tower client will behave today as well. 

I also ran into what might be an issue. I'll update this when I have a bit more time on my hands.